envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/permissions/IncomingPermissionsRequestIdTest.kt

```diff
@@ -0,0 +1,103 @@
+import pl.allegro.tech.servicemesh.envoycontrol.assertions.hasOneAccessDenialWithActionLog
+import pl.allegro.tech.servicemesh.envoycontrol.assertions.isOk
+import pl.allegro.tech.servicemesh.envoycontrol.assertions.isRbacAccessLog
+import pl.allegro.tech.servicemesh.envoycontrol.assertions.untilAsserted
+import pl.allegro.tech.servicemesh.envoycontrol.config.Echo1EnvoyAuthConfig
+import pl.allegro.tech.servicemesh.envoycontrol.config.consul.ConsulExtension
+import pl.allegro.tech.servicemesh.envoycontrol.config.envoy.EnvoyExtension
+import pl.allegro.tech.servicemesh.envoycontrol.config.envoycontrol.EnvoyControlExtension
+import pl.allegro.tech.servicemesh.envoycontrol.config.service.EchoServiceExtension
+
+class IncomingPermissionsRequestIdTest {
+
+    companion object {
+
+        // language=yaml
+        private val echoYaml = """
+            node:
+              metadata:
+                proxy_settings:
+                  incoming:
+                    endpoints:
+                    - path: "/path"
+                      unlistedClientsPolicy: log
+                      clients: []
+                  outgoing:
+                    dependencies:
+                      - service: "echo"
+        """.trimIndent()
+
+        private val echoConfig = Echo1EnvoyAuthConfig.copy(configOverride = echoYaml)
+
+        @JvmField
+        @RegisterExtension
+        val consul = ConsulExtension()
+
+        @JvmField
+        @RegisterExtension
+        val envoyControl = EnvoyControlExtension(consul, mapOf(
+            "envoy-control.envoy.snapshot.incoming-permissions.enabled" to true,
+            "envoy-control.envoy.snapshot.incoming-permissions.request-identification-headers" to listOf("x-request-id")
+        ))
+
+        @JvmField
+        @RegisterExtension
+        val echoService = EchoServiceExtension()
+
+        @JvmField
+        @RegisterExtension
+        val echoEnvoy = EnvoyExtension(envoyControl, config = echoConfig, localService = echoService)
+    }
+
+    @BeforeEach
+    fun beforeEach() {
+        consul.server.operations.registerServiceWithEnvoyOnIngress(echoEnvoy, name = "echo")
+        echoEnvoy.container.logRecorder.recordLogs(::isRbacAccessLog)
+    }
+
+    @AfterEach
+    fun afterEach() {
+        echoEnvoy.container.logRecorder.stopRecording()
+    }
+
+    @Test
+    fun `incoming permissions logs should contain requestId if present`() {
+        untilAsserted {
+            // when
+            val response = echoEnvoy.egressOperations.callService(service = "echo", pathAndQuery = "/path", headers = mapOf(
+                "x-request-id" to "123"
+            ))
+
+            // then
+            assertThat(response).isOk()
+            assertThat(echoEnvoy.container).hasOneAccessDenialWithActionLog(
+                protocol = "http",
+                requestId = "123"
+            )
+        }
+    }
+
+    @Test
+    fun `should handle request id containing double quotes`() {
+        untilAsserted {
```

Comment:
I think it's a bad practice to test condition in `untilAsserted` block. Let's wait for envoys to be ready in beforeAll phase, then run the tests only once. Just like we do in recent tests: https://github.com/allegro/envoy-control/blob/master/envoy-control-tests/src/main/kotlin/pl/allegro/tech/servicemesh/envoycontrol/permissions/ClientNameTrustedHeaderTest.kt#L101

Running test in `untilAsserted` block makes it weak - if for some reason / bug in the code `x-request-id` is set only for 1 of 10 requests, the test will still pass.